sycl/include/CL/sycl/pipes.hpp

```diff
@@ -29,7 +51,9 @@ template <class name, class dataT, int32_t min_capacity = 0> class pipe {
         __spirv_ReadPipe(RPipe, &TempData, m_Size, m_Alignment));
     return TempData;
 #else
-    assert(!"Pipes are not supported on a host device!");
+    __pipe_printer<name>("Non-blocking read data from pipe: ", m_Size,
+                         m_Alignment, m_Capacity);
+    return dataT();
```

Comment:
The title "Add host debugging for pipes" implies that you can now debug pipes on a host.
I'm pretty sure that no sensible program can be debugged without an actual data flow.

If you want to support host debugging for pipes, then you need to implement this as well.